macros.py

```diff
@@ -831,20 +840,35 @@ def overall_graph():
                 graph_file.write(line)
         graph_file.write("}\n")
 
-def month_graphs(dates):
+def month_graphs(dates, version=None):
     """Produce a graph per month, showing the exploits that were possible on the first day of that month"""
+    version_string = version
+    if version == None:
+        version_string = 'all'
+    print("Graphing version " + version_string)
+    # Create a folder for graphs of this version, if it doesn't exist already
+    folder_path = 'graphs/' + version_string.replace('.', '-')
+    pathlib.Path(folder_path).mkdir(exist_ok=True)
+    # Flag so that there aren't lots of blank graphs from before this version was ever exploited
+    yet_found = False
     for date in dates:
-        filename = 'graphs/graph-{:d}-{:02d}.gv'.format(date.year, date.month)
-        with open(filename, 'w') as graph_file:
-            graph_file.write('digraph vulnerabilities {\n')
-            for vulnerability in vulnerabilities:
-                if vulnerability.exploitable_on(date):
+        exploitables = []
+        for vulnerability in vulnerabilities:
+            if vulnerability.exploitable_on(date, version):
+                if version == None or vulnerability.regex().match(version):
+                    yet_found = True
                     for line in vulnerability.graphLines(condition_privilege_lookup):
-                        graph_file.write(line)
-            graph_file.write('labelloc="top";\n')
-            graph_file.write('labeljust="right";\n')
-            graph_file.write('label="{:d}-{:02d}";\n'.format(date.year, date.month))
-            graph_file.write('}\n')
+                        exploitables.append(line)
+        if yet_found:
+            filename = folder_path + '/graph-{:d}-{:02d}.gv'.format(date.year, date.month)
+            with open(filename, 'w') as graph_file:
+                graph_file.write('digraph vulnerabilities {\n')
+                for line in exploitables:
+                    graph_file.write(line)
+                graph_file.write('labelloc="top";\n')
```

Comment:
Perhaps put the `version` in the top left corner?

Reply:
As discussed - now added to the caption